### PR TITLE
FHG-5241 Metric DTOs

### PR DESCRIPTION
**Pull request by AaronJY**
- Adds DTOs for metrics
- Adds `DS_Store` to `.gitignore`
- Increments to `8.9.0`

### CODE REVIEW

File: src/FamilyHubs.ServiceDirectory.Shared/Dto/Metrics/EventDto.cs [hunk 1]
```diff
@@ -0,0 +1,10 @@
+using FamilyHubs.ServiceDirectory.Shared.Enums;
+
+namespace FamilyHubs.ServiceDirectory.Shared.Dto.Metrics;
+
+public class EventDto
+{
+    public ServiceDirectorySearchEventType? Id { get; set; }
```
Comment (Top-Cat) [inline src/FamilyHubs.ServiceDirectory.Shared/Dto/Metrics/EventDto.cs]: Is this an Id?

Comment (AaronJY) [reply src/FamilyHubs.ServiceDirectory.Shared/Dto/Metrics/EventDto.cs]: Yes, driven by the enum `ServiceDirectorySearchEventType`. It's a lookup table for categorising metric event types. EF seeds the data with two IDs in the app context, and the enum maps to those IDs. This means we can use the enum to refer to specific event types throughout the system :)

Comment (AaronJY) [reply src/FamilyHubs.ServiceDirectory.Shared/Dto/Metrics/EventDto.cs]: Just for context, this is what the table looks like:

Screenshot 2024-05-07 at 16 15 34

**Review by Top-Cat — APPROVED** [review]

File: .gitignore [hunk 2]
```diff
@@ -352,4 +352,6 @@ MigrationBackup/
 # Ignore sonarqube files
 .sonarqube/
 
-*.idea*
\ No newline at end of file
+*.idea*
```
Comment (Zac-Digital) [inline .gitignore]: We should keep the `.idea` folder for shared config purposes, they contain their own `.gitignore` files to filter out any unneeded stuff :)

File: src/FamilyHubs.ServiceDirectory.Shared/Dto/Metrics/ServiceSearchDto.cs [hunk 3]
```diff
@@ -0,0 +1,19 @@
+using FamilyHubs.ServiceDirectory.Shared.Enums;
+
+namespace FamilyHubs.ServiceDirectory.Shared.Dto.Metrics;
+
+public class ServiceSearchDto
+{
+    public long? Id { get; set; }
```
Comment (Zac-Digital) [inline src/FamilyHubs.ServiceDirectory.Shared/Dto/Metrics/ServiceSearchDto.cs]: Should `Id` be nullable here? Same with `CorrelationId` below

File: src/FamilyHubs.ServiceDirectory.Shared/Dto/Metrics/ServiceSearchResultDto.cs [hunk 4]
```diff
@@ -0,0 +1,8 @@
+namespace FamilyHubs.ServiceDirectory.Shared.Dto.Metrics;
+
+public class ServiceSearchResultDto
+{
+    public long? Id { get; set; }
```
Comment (Zac-Digital) [inline src/FamilyHubs.ServiceDirectory.Shared/Dto/Metrics/ServiceSearchResultDto.cs]: Same question as above, `Id` & `ServiceSearchesId` being nullable :)

**Review by Zac-Digital — CHANGES_REQUESTED** [review]
Just a few little comments!

File: .gitignore (hunk 2 above)
Comment (AaronJY) [reply .gitignore]: `.idea` was already ignored, but happy to ignore it here if we want to keep it!

**Review by Zac-Digital — APPROVED** [review]

**Pull request merged**